_posts/2022-12-19-tm1-project.md

```diff
@@ -0,0 +1,222 @@
+| :---------------------------------------: | :----------------------------------------------------------: |
+| ability to single out changes and commits |    the entire codebase gets staged, committed and pushed     |
+|        can mirror repo after pull         |                does not delete files on disk                 |
+|    repo and local have the same files     | repo has a representation of local file (i.e.: local: ".pro" => remote: ".json" + ".ti") |
+
+If you are unfamiliar with tm1-git or you've never seen it in action,  I would strongly suggest you to go over these 2 articles:
+
+- [GIT integration for TM1, part 1](https://www.ykud.com/blog/cognos/tm1-cognos/git-integration-for-tm1-part-1/) 
+- [GIT integration for TM1, part 2](https://www.ykud.com/blog/cognos/tm1-cognos/git-integration-for-tm1-part-2/)
+
+They provide step by step instructions on how you can setup and interact with a Git repository fro scratch. Bare in mind that all the REST API functions shown there are now availabe in TM1Py (as of relase [1.10](https://github.com/cubewise-code/tm1py/releases/tag/1.10.0))
+
+### What is a tm1project?
+
+A tm1project is where you can define how a TM1 model should be published (pushed) and deployed (pulled). 
+You can specify the objects that get published as well as the tasks that should be executed before, during and after a deployment. You can think of it as the place you setup your strategy towards code promotion. You can review [IBM Documentation on tm1project](https://www.ibm.com/docs/en/planning-analytics/2.0.0?topic=git-tm1-model-source-specification) for further details.
+
+### Scope and limitations
+
+As mentioned above in the Git vs TM1-Git table, there are some limitaitons that you need to be aware of before trying to use this approach:
+
+- understand that tm1-git (with or without a tm1project) is only intended publish the structre of a TM1 model, **NOT** the data inside of it.
+- tm1-git does not remove objects from the tm1server (even if they were deleted from the source branch), this is by design.
+- physical dimension order is **NOT** part of the published cube object for TM1 v11 [request](https://ibm-data-and-ai.ideas.ibm.com/ideas/PAOC-I-955)  
+- tm1-git does **NOT** allow you to commit selected objects, anything that has changed since last commit will be automatically staged and committed to the repo on your next push. You need to remove the unwanted objects from the branch to which they were committed to.
+
+## How could you use it?
+
+Considering the limitations that this functionality has you should probably stablish some ground rules amongst your development team to properly work with it:
+
+1. Use an 'official' tm1project (like the one below) so everybody know what gets pushed/pulled and what happens before and after each action. If you change it everytime just to push what you want it will become unpredictable. Build *around* it, **not** in it.
```

Comment:
`1. Use an 'official' tm1project (like the one below) so everybody know what gets pushed/pulled and what happens before and after each action. If you change it everytime just to push what you want it will become unpredictable. Build *around* it, **not** in it.`

Well put!